src/nupic/algorithms/TemporalMemory.hpp

```diff
@@ -463,9 +503,11 @@ namespace temporal_memory {
   Permanence permanenceIncrement_;
   Permanence permanenceDecrement_;
   Permanence predictedSegmentDecrement_;
+  UInt extra_;
```

Comment:
serialization & equals have to change for extra_, and possibly also segmentsValid_